mesh.js

```diff
@@ -0,0 +1,455 @@
+                    mesh.texcoordsAccessor = accessor;
+                    break;
+                case "INDEX": mesh.indices = data;
+                    mesh.indicesAccessor = accessor;
+                    break;
+                default:
+                    console.warn('Unknown attribute semantic: ' + attribute);
+            }
+
+            mesh.accessorsLoading--;
+            if (mesh.accessorsLoading === 0) {
+                mesh.initBuffers(gl, gltf);
+            }
+        };
+
+        var assets = mesh.scene.assets;
+        var assetUrl = modelPath + bin;
+        var promise;
+        if (assets.hasOwnProperty(assetUrl)) {
+            // We already requested this, and a promise already exists.
+            promise = assets[assetUrl];
+        } else {
+            // We didn't request this yet, create a promise for it.
+            var deferred = $.Deferred();
+            assets[assetUrl] = deferred;
+            promise = deferred.promise();
+            var oReq = new XMLHttpRequest();
+            oReq.open("GET", assetUrl, true);
+            oReq.responseType = "blob";
+            oReq.onload = function(e) {
+                deferred.resolve(oReq.response);
+            };
+            oReq.send();
+        }
+
+        // This will fire when the promise is resolved, or immediately if the promise has previously resolved.
+        promise.then(function(blob) {
+            reader.readAsArrayBuffer(blob);
+        });
+    }
+}
+
+//
+// **** NOTE: The following functions are global ****
+//
+
+function loadImage(imageInfo, gl, mesh) {
+    var image = new Image();
+    mesh.scene.pendingTextures++;
+    image.src = imageInfo.uri;
+    image.onload = function() {
+        var texture = gl.createTexture();
+        var glIndex = gl.TEXTURE0 + imageInfo.samplerIndex;  // gl.TEXTUREn enums are in numeric order.
+        gl.activeTexture(glIndex);
+        gl.bindTexture(gl.TEXTURE_2D, texture);
+
+        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, imageInfo.clamp ? gl.CLAMP_TO_EDGE : gl.REPEAT);
+        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, imageInfo.clamp ? gl.CLAMP_TO_EDGE : gl.REPEAT);
+        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.LINEAR);
+        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.LINEAR);
+        gl.pixelStorei(gl.UNPACK_FLIP_Y_WEBGL, false);
+        gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA,/*imageInfo.colorSpace, imageInfo.colorSpace,*/ gl.UNSIGNED_BYTE, image);
+
+        mesh.scene.pendingTextures--;
+
+        if (mesh.loadedBuffers === true && mesh.scene.pendingTextures === 0) {
+            mesh.scene.drawScene(gl);
+        }
+    };
+
+    return image;
+}
+
+function loadImages(imageInfos, gl, mesh) {
+    mesh.scene.pendingTextures = 0;
+    for (var i in imageInfos) {
+        loadImage(imageInfos[i], gl, mesh);
+    }
+}
+
+function applyState(gl, program, globalState, localState) {
```

Comment:
Unless we want to add a context.js (not suggesting this), I'd just move this into the Mesh class too.